R/derive_vars_query.R

```diff
@@ -61,14 +61,37 @@
 #'   7, "Alveolar proteinosis", NA_character_, NA_integer_
 #' )
 #' derive_vars_query(adae, queries)
-derive_vars_query <- function(dataset, dataset_queries) {
+derive_vars_query <- function(dataset, dataset_queries) { # nolint: cyclocomp_linter
   assert_data_frame(dataset_queries)
   assert_valid_queries(dataset_queries, queries_name = deparse(substitute(dataset_queries)))
   assert_data_frame(dataset,
     required_vars = exprs(!!!syms(unique(dataset_queries$SRCVAR))),
```

Comment:
Should we define `source_vars <- unique(dataset_queries$SRCVAR))` then we could use it here and in the new check.

We could also use `chr2var()` instead of `exprs(!!!syms())`.